### ISSUE TITLE
🐛 `amp-story-interactive` accessibility issues

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amp.dev#5400]
### Current behavior
While `<amp-story-interactive>` was not part of our recent accessibility assessment, we noticed (while going through the various parts of the documentation) that these have some serious fundamental issues:

- none of the controls for quizzes (the buttons for binary polls or interactive polls) are keyboard-accessible - they can't be focused nor selected when using a keyboard
- the polls themselves and the various options are not announced as being actionable by assistive technologies
- when polls update to show results, this change is not announced by assistive technologies

Based on briefly going over https://amp.dev/documentation/components/amp-story-interactive/?format=stories and the demos at https://amp.dev/documentation/examples/components/amp-story-interactive-quiz/preview/?format=stories and https://amp.dev/documentation/examples/components/amp-story-interactive-poll/preview/?format=stories

### Expected behavior/code
The way these components are rendered should be reviewed. The individual options need to be keyboard-accessible - suggest using something like a `<button>` element for each of these. The entirety of the quiz/poll should be wrapped in a `<fieldset>`, and the overarching question marked as a `<legend>`.

Changes that happen dynamically (e.g. results being shown) should use an ARIA live region to ensure they are announced by assistive technologies.

### Reproduction procedure
Please provide the steps to reproduce the problem:

1. Navigate to either https://amp.dev/documentation/examples/components/amp-story-interactive-quiz/preview/?format=stories or https://amp.dev/documentation/examples/components/amp-story-interactive-poll/preview/?format=stories
2. Attempt to complete the poll/quiz using a keyboard (`Tab` / `Shift`+`Tab` to move focus, `Space` or `Enter` to activate a button/option

- Chrome / Windows
